src/customFileParser/main.ts

```diff
@@ -0,0 +1,81 @@
+  from './sharedFileParserTypes';
+import CustomFileReader from './CustomFileReader';
+import { renderedGeometries, renderedGeometryToGroup } from './parserParameters';
+
+function fillSector(
+  geometryIndexHandlers: GeometryIndexHandler[],
+  sectorMetadata: SectorMetadata,
+  trueValueArrays: TrueValueArrays): Sector {
+
+  const counts = countRenderedGeometries(geometryIndexHandlers);
+  const renderedGeometryGroups: RenderedGeometryGroups = {};
+  renderedGeometries.forEach(renderedGeometry => {
+    const count = counts[renderedGeometry];
+    renderedGeometryGroups[renderedGeometry] = new renderedGeometryToGroup[renderedGeometry](count);
+  });
+
+  const groupNames = ['Box', 'Circle', 'ClosedCone', 'ClosedCylinder', 'ClosedEccentricCone', 'ClosedEllipsoidSegment',
+    'ClosedExtrudedRingSegment', 'ClosedGeneralCylinder', 'ClosedSphericalSegment', 'ClosedTorusSegment', 'Ellipsoid',
+    'ExtrudedRing', 'Nut', 'OpenCone', 'OpenCylinder', 'OpenEccentricCone', 'OpenEllipsoidSegment',
+    'OpenExtrudedRingSegment', 'OpenGeneralCylinder', 'OpenSphericalSegment', 'OpenTorusSegment', 'Ring', 'Sphere',
+    'Torus', 'TriangleMesh', 'InstancedMesh'];
+  groupNames.forEach(name => {
+    unpackGeometry(renderedGeometryGroups, geometryIndexHandlers, trueValueArrays, name);
+  });
+
+  const sectorObject = new Sector(sectorMetadata.sectorBBoxMin, sectorMetadata.sectorBBoxMax);
+  Object.keys(renderedGeometryGroups).forEach(renderedGeometryName => {
+    if (renderedGeometryName === 'TriangleMesh') {
+
+    } else if (renderedGeometryName === 'InstancedMesh') {
+
+    } else {
+      sectorObject.primitiveGroups.push(renderedGeometryGroups[renderedGeometryName]);
+    }
+  });
+
+  return sectorObject;
+}
+
+function parseManySectors(incomingFile: ArrayBuffer): Sector {
+  const file = new CustomFileReader(incomingFile);
+  const sectors: {[name: string]: any} = {};
+
+  let sectorStartLocation = file.location;
+  let sectorByteLength = file.readUint32();
+  let sectorMetadata = file.readSectorMetadata(sectorByteLength);
+  const trueValueArrays = file.readTrueValueArrays();
+  let geometryIndexHandlers = file.readSectorGeometryIndexHandlers(sectorStartLocation + sectorByteLength);
+  const rootSector = fillSector(geometryIndexHandlers, sectorMetadata, trueValueArrays);
+  sectors[sectorMetadata.sectorId] = rootSector;
+
+  while (file.location !== incomingFile.byteLength) {
+    sectorStartLocation = file.location;
+    sectorByteLength = file.readUint32();
+    sectorMetadata = file.readSectorMetadata(sectorByteLength);
+    geometryIndexHandlers = file.readSectorGeometryIndexHandlers(sectorStartLocation + sectorByteLength);
+    const newSectorObject = fillSector(geometryIndexHandlers, sectorMetadata, trueValueArrays);
+    const parentSector = sectors[sectorMetadata.parentSectorId];
+    if (parentSector !== undefined) {
+      parentSector.addChild(newSectorObject);
+      parentSector.object3d.add(newSectorObject.object3d);
```

Comment:
Why do you have the same `object3d` in both the child and the parent? They should have different `THREE.Object3d` objects.